app/models/server/models/Messages.js

```diff
@@ -1260,6 +1260,19 @@ export class Messages extends Base {
 		return this.find(query);
 	}
 
+	findLivechatClosedMessages(rid, options) {
+		return this.find(
+			{
+				rid,
+				$or: [
+					{ t: { $exists: false } },
+					{ t: 'livechat-close' },
```

Comment:
```suggestion
```
This condition has no effect because of the previous OR condition i.e. `t: { $exists: false }`, so I think we can safely remove it. In fact, this entire OR condition isn't required. we can simply keep the filter as 
```
{
     rid,
     t: { $exists: false },
}
```
and this should give you only the text messages for this room